ldap/servers/plugins/acl/acl.c

```diff
@@ -2408,10 +2469,53 @@ acl__resource_match_aci(Acl_PBlock *aclpb, aci_t *aci, int skip_attrEval, int *a
                                                     ACL_EVAL_TARGET_FILTER);
             slapi_ch_free((void **)&lasinfo);
         } else {
-            if (slapi_vattr_filter_test(NULL, aclpb->aclpb_curr_entry,
-                                        aci->targetFilter,
-                                        0 /*don't do access check*/) != 0) {
-                filter_matched = ACL_FALSE;
+            Slapi_DN *sdn;
+            char* attr_evaluated = "None";
+            char logbuf[1024];
```

Comment:
Is this big enough?  What happens if the filter is longer than 1024 characters?

Reply:
I guess it's a corner case, since a target filter in an ACI would not typically be very big, but you never know.

Reply:
I agree, I changed that so that if the filterstr gets truncated then it is ignored and we systematically reevaluate the targetfilter

Reply:
I guess the other factor there is if the filter is really large then those are going to be more expensive and that's really where we would make gains in performance?

Reply:
I see your point: The larger is the targetfilter the more beneficial should be to cache its result.
On the other side I did not want to loop allocate buffer/convert filter/check filterstr is truncated because of performance impact. So using a static buffer was the most immediate solution as most of the targetfilters will fit into.
I see 2 options
-  loop allocate/convert/check
- use a static buffer and if the targetfilter is too large to fit into it then log an info that the cache is not used for this targetfilter.

Reply:
Instead of looping allocating buffer/testing overwrite until the buffer is large enough, the modified fix keeps a static buffer and detect overwrite (redzone). In case of overwrite the filter is just ignore (does not go into the cache).

Also the buffer is not 2K.